pyxem/utils/expt_utils.py

```diff
@@ -260,12 +260,18 @@ def apply_transformation(z, transformation,keep_dtype,order=1,*args, **kwargs):
     Generally used in combination with pyxem.expt_utils.convert_affine_to_transform
     """
     if keep_dtype == False:
-        trans = tf.warp(z, transformation,
-                    order=order, *args, **kwargs)
+        if z._lazy == True:
+            trans = z.data.map_blocks(tf.warp, transformation,order = order, *args, **kwargs)
```

Comment:
Why not using `z.map(...)`, so there is no need to worry about the signal being lazy.

Reply:
Looking the hyperspy function hyperspy.signals.map, it does not appear to operate lazily. This is obviously a problem with the large 4DSTEM data sets. Please correct me if I'm wrong as simply using z.map(...) would be more straightforward.

Reply:
`map` supports lazy signal (see https://github.com/hyperspy/hyperspy/blob/63bd4a2960b0dbeb4b1e5fbf82ecfd48bc6e65d9/hyperspy/_signals/lazy.py#L481) and as @dnjohnstone mentioned, it is already using `map` https://github.com/pyxem/pyxem/blob/ba0ed6934f00fe09435c5455d90247815b3fe758/pyxem/signals/diffraction2d.py#L203
so in principle it should be already working! If not there should be bug somewhere.

Reply:
Indeed I think this should be working in pyxem-0.9.0 - we just didn't have any LazySignal classes until last week - but anticipating this we tried quite hard to ensure the hyperspy map function was used wherever possible, which as you say should mean that the vast majority of existing features work out of the box in the lazy case.

I think we'll iron this out over the lifetimes of v0.9.0 (anticipated to be until the end of the summer) and evaluate more carefully what we actually need to do (if anything) for lazy signals.